kubeadm: improvements to the cert management documentation (#18397)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md

{{% capture steps %}}

## Using custom certificates {#custom-certificates}

By default, kubeadm generates all the certificates needed for a cluster to run.
You can override this behavior by providing your own certificates.

To do so, you must place them in whatever directory is specified by the
`--cert-dir` flag or the `certificatesDir` field of kubeadm's `ClusterConfiguration`.
By default this is `/etc/kubernetes/pki`.

If a given certificate and private key pair exists before running `kubeadm init`,

> kubeadm will not overwrite them. This means you can, for example, copy an existing

Reviewer: style mini-nit:
```suggestion
kubeadm does not overwrite them. This means you can, for example, copy an existing
```

CA into `/etc/kubernetes/pki/ca.crt` and `/etc/kubernetes/pki/ca.key`,
and kubeadm will use this CA for signing the rest of the certificates.

## External CA mode {#external-ca-mode}

It is also possible to provide just the `ca.crt` file and not the
`ca.key` file (this is only available for the root CA file, not other cert pairs).
If all other certificates and kubeconfig files are in place, kubeadm recognizes
this condition and activates the "External CA" mode. kubeadm will proceed without the
CA key on disk.

Instead, run the controller-manager standalone with `--controllers=csrsigner` and
point to the CA certificate and key.

A more detailed guide about setting up external CA can be found

> [here](https://kubernetes.io/docs/setup/best-practices/certificates/).

Reviewer: This is different from the style guide, and also the absolute URI should become site-relative.

How about:
```suggestion
[PKI certificates and requirements](/docs/setup/best-practices/certificates/) includes guidance on
setting up a cluster to use an external CA>
```

Author: typo - `>` instead of `.` in suggestion.
i will update it.

## Check certificate expiration

> The `check-expiration` command can be used to check certificate expiration.

Reviewer: style nit:
```suggestion
You can use the `check-expiration` subcommand to check when certificates expire:
```
or
```suggestion
You can use kubeadm to check when certificates expire:
```

Also: mention alpha status?

Reviewer: Bear in mind that the overall page is marked stable:
```markdown
{{< feature-state for_k8s_version="v1.15" state="stable" >}}
```

Author: the command is already denoted as alpha in the shell example.
